auslib/db.py

```diff
@@ -2098,6 +2096,30 @@ def _proceedIfNotReadOnly(self, name, limit=None, transaction=None):
         if self.isReadOnly(name, limit, transaction):
             raise ReadOnlyError("Release '%s' is read only" % name)
 
+    def change_readonly(self, where, is_readonly, changed_by, old_data_version, transaction=None, dryrun=False, signoffs=None):
+        release = self.select(where=where, columns=[self.name, self.product], transaction=transaction)[0]
+        product = release["product"]
+
+        if is_readonly:
+            if not self.db.hasPermission(changed_by, "release_read_only", "set", product, transaction):
+                raise PermissionDeniedError(f"{changed_by} is not allowed to mark {product} products read only")
+        else:
+            if not self.db.hasPermission(changed_by, "release_read_only", "unset", product, transaction):
+                raise PermissionDeniedError(f"{changed_by} is not allowed to mark {product} products read write")
+
+        if not dryrun and not is_readonly:
+
+            def _getPotentialRequiredSignoffs(required_signoffs):
```

Comment:
AFAICT this function just transforms the data in `required_signoffs`. It could use a better name, perhaps relating to what it actually returns.